pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationProvider.java

```diff
@@ -185,6 +185,18 @@ CompletableFuture<Boolean> allowSinkOpsAsync(NamespaceName namespaceName, String
     CompletableFuture<Void> grantPermissionAsync(NamespaceName namespace, Set<AuthAction> actions, String role,
             String authDataJson);
 
+    /**
+     * Revoke authorization-action permission on a namespace to the given client.
+     * @param namespace
+     * @param role
+     * @return
+     */
+    default CompletableFuture<Void> revokePermissionAsync(NamespaceName namespace, String role) {
```

Comment:
Why do we need a default implementation here? How about removing it so that we can keep aligned with other APIs in this file?

Reply:
It’s better to have `default` implementations here,  otherwise it will cause compile issue when upgrade.

Reply:
Agree, also the "revoke" is already broken today when using a custom authz plugin. Even if today it succeeds, the revoke does nothing because it would just try to remove the ACL from zookeeper, where there are not any.